buildkite/resource_pipeline_schedule_test.go

```diff
@@ -0,0 +1,201 @@
+package buildkite
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+// Confirm that we can add a new pipeline schedule to a pipeline
+func TestAccPipelineSchedule_add_remove(t *testing.T) {
+	var resourcePipeline PipelineNode
+	var resourceSchedule PipelineScheduleNode
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckPipelineScheduleResourceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccPipelineScheduleConfigBasic("foo", "0 * * * *"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Schedules need a pipeline
+					testAccCheckPipelineExists("buildkite_pipeline.foobar", &resourcePipeline),
+					// Confirm the schedule exists in the buildkite API
+					testAccCheckPipelineScheduleExists("buildkite_pipeline_schedule.foobar", &resourceSchedule),
+					// Confirm the schedule has the correct values in Buildkite's system
+					testAccCheckPipelineScheduleRemoteValues(&resourcePipeline, &resourceSchedule, "main", "Test Schedule foo", "0 * * * *"),
+					// Confirm the pipeline has the correct values in terraform state
+					resource.TestCheckResourceAttr("buildkite_pipeline_schedule.foobar", "label", "Test Schedule foo"),
```

Comment:
is it worth checking the state has other properties set correctly as well (like cronline and branch)